simple-ipc.c

```diff
@@ -31,6 +31,22 @@ static int initialize_pipe_name(const char *path, wchar_t *wpath, size_t alloc)
 	return 0;
 }
 
+static int is_active(wchar_t *pipe_path)
```

Comment:
Now that we are implementing these features simultaneously in both platforms, I wonder if we should be changing the file structure a bit. It is hard to tell from the patch context that we are in the `#ifdef WINDOWS` region.

Perhaps we create an `ipc` directory, and have a split of files this way:

```
ipc/
    simple-ipc.h
    simple-ipc-internal.h (stores struct and method prototypes used by both *-ipc.c files)
    simple-ipc.c  (stores platform-agnostic logic)
    windows-ipc.c
    unix-ipc.c
```

For example, the different platform-specific pieces of `ipc_listen_for_commands()` could be declared in `simple-ipc-internal.h` and implemented in `<platform>-ipc.c`, but `ipc_listen_for_commands()` itself is declared in `simple-ipc.h` and implemented in `simple-ipc.c`.

Then, the `Makefile` could decide which of `windows-ipc.c` or `unix-ipc.c` to compile and link.

I know it would be a lot of work to rearrange this, but do you think this is a better direction for long term maintainability?

Reply:
I don't know that this would make it easier to read. I like to have the IPC things in one file, and it strikes me as a similar situation to `run-command.c`, where large parts are `#ifdef GIT_WINDOWS_NATIVE ... #else ... #if`. It's just that C preprocessor idioms cause this kind of readability issues, but that's not something I can change in this PR.